contracts/modules/TransferManager/VolumeDumpingRestrictionManager.sol

```diff
@@ -0,0 +1,326 @@
+        uint256 _startTime, 
+        uint256 _endTime, 
+        uint256 _rollingPeriod
+        ) public withPerm(ADMIN) {
+
+        _checkVolumeDumpingParams(_userAddress, _percentAllowed, _startTime, _endTime, _rollingPeriod);
+ 
+        require(volumeRestriction[_userAddress].endTime > now, "Cannot modify past restrictions");
+
+        if (_startTime == 0) {
+            _startTime = now;
+        }
+
+        require(_endTime > _startTime, "End time must be greater than start time");
+
+        volumeRestriction[_userAddress] = VolumeRestriction(_percentAllowed, _startTime, _endTime, _rollingPeriod);
+
+        emit ModifyVolumeDumping(
+            _userAddress,
+            _percentAllowed,
+            _startTime,
+            _endTime,
+            _rollingPeriod
+        );
+    }
+    
+    /**
+    * @notice Get volume dumping restriction rules for a user address
+    * @param _userAddress User Address to get the rule
+    */
+    function getVolumeDumpingRestrictions(address _userAddress) public view returns (
+        uint256 _percentAllowed , uint256 _startTime, uint256 _endTime, uint256 _rollingPeriod
+    ) {
+        VolumeRestriction storage userRestriction = volumeRestriction[_userAddress];
+        return (
+            userRestriction.percentAllowed,
+            userRestriction.startTime,
+            userRestriction.endTime,
+            userRestriction.rollingPeriod
+        );
+    }
+
+    /**
+     * @notice This function returns the signature of configure function
+     */
+    function getInitFunction() public pure returns (bytes4) {
+        return bytes4(0);
+    }
+
+
+    /**
+     * @notice Return the permissions flag that are associated with Volume Restriction transfer manager
+     */
+    function getPermissions() public view returns(bytes32[]) {
+        bytes32[] memory allPermissions = new bytes32[](1);
+        allPermissions[0] = VOLUME_RESTRICTION;
+        return allPermissions;
+    }
+
+     /**
+    * @notice Takes a userAddress as input, and returns a uint that represents the number of tokens allowed to be withdrawn right now
+    * @param _userAddress User Address to get the rule
+    * @param periodId Time period identifier
+    */
+    function _currentAllowedAmount(address _userAddress, uint periodId) internal returns (uint) {
+        VolumeRestriction storage userDumpingRestriction = volumeRestriction[_userAddress];
+        
+        uint alreadyWithdrawn = volumeTally[_userAddress][periodId];
+        // if the already with drawn is 0
+        // means its essentially a new period
+        if(alreadyWithdrawn == 0){
+            volumePeriodIds[_userAddress].push(periodId);
+        }
+
+        uint256 currentUserBalance = ISecurityToken(securityToken).balanceOf(_userAddress);
+        // add already withdrawn amount for this period 
+        // to be able to get the correct value
+
+        uint256 percentAllowed = userDumpingRestriction.percentAllowed;
```

Comment:
For better accuracy percentages should be represented as [percentage] * 10**16. So 100% == 10**18, 50% = 0.5 * 10**18